### PR TITLE
Clean up some documentation debt

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

File: docs/user/features.md [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+
+- Import path
+
+```rust
+// before:
+impl std::fmt::Debug<|> for Foo {
+}
+
+// after:
+use std::fmt::Debug
```
Comment (flodiebold) [inline docs/user/features.md]: `;` missing here

File: docs/user/features.md [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+    a: u32,
+    b: String,
+    c: (i32, i32),
+    d: D,
+    r: &'a str,
+}
+
+fn main() {
+    let s = <|>S {
+        a: (),
```
Comment (flodiebold) [inline docs/user/features.md]: hmm, not related to the documentation, but shouldn't these be replaced by `unimplemented!()` / `todo!()` / some good defaults?

Comment (matklad) [reply docs/user/features.md]: That's a good question!

Defaults are problematic (except for field shorthand, which should be done): you can't say if you intended to write 0 or if you've just forgotten to fill the field.

`unimplemented!()` is long and noisy. Specifically, it's a pain to delete, if you don't know about extend selection. `()` is trivial to delete.

**Comment by killercup** [comment]
> docs/dev/arhictecture.md

File: README.md [hunk 3]
```diff
@@ -13,116 +13,44 @@ Work on the Rust Analyzer is sponsored by
...
 
-```
-# run tests
-$ cargo test
+To build rust-analyzer, you need:
 
-# show syntax tree of a Rust file
-$ cargo run --package ra_cli parse < crates/ra_syntax/src/lib.rs
+* latest stable rust for language server itself
+* latest stable npm and VS Code for VS Code extension (`code` should be a path)
```
Comment (bjorn3) [inline README.md]:
```suggestion
* latest stable npm and VS Code for VS Code extension (`code` should be in path)
```

**Comment by Xanewok** [comment]
That’s a definite improvement! Not sure how I feel about dropping the somewhat customary CONTRIBUTING (maybe we could link from there to /dev/README?) but I don’t feel strongly about it, so:

bors r+

**Comment by kjeremy** [comment]
I love how this is PR 1000

**Pull request merged**